src/app/home/home.component.ts

```diff
@@ -28,7 +34,29 @@ export class HomeComponent implements OnDestroy {
     }
   }
 
+  ngOnInit() {
+    this.videoQualitySelect = document.querySelector('#video-quality')! as HTMLSelectElement;
+    this.videoCameraSelect = document.querySelector('#video-camera')! as HTMLSelectElement;
+    this.videoQualitySelect.value =
+      localStorage.getItem('defaultQuality') || '480';
+    this.videoCameraSelect.value =
+      localStorage.getItem('defaultCamera') || 'user';
+  }
+
   ngOnDestroy() {
     this.subsink.unsubscribe();
   }
+
+  onClick() {
+    let quality = this.videoQualitySelect!.options[this.videoQualitySelect!.selectedIndex].value;
+    let camera = this.videoCameraSelect!.options[this.videoCameraSelect!.selectedIndex].value;
+    localStorage.setItem('defaultQuality', quality);
+    localStorage.setItem('defaultCamera', camera);
+    this.router.navigate(['/run'], {
+      queryParams: {
+        h: quality,
+        cam: camera,
```

Comment:
How about take these out of the URL query params and just read from localstorage in the /run page?

Reply:
Done.